include/deal.II/base/vectorization.h

```diff
@@ -1148,8 +1148,9 @@ class VectorizedArray<double, 8>
     // work around a warning with gcc-12 about an uninitialized initial state
     // for gather by starting with a zero guess, even though all lanes will be
     // overwritten
-    __m512d  zero = {};
-    __mmask8 mask = 0xFF;
+    __m512d       zero    = {};
+    const __m256i invalid = _mm256_set1_epi32(numbers::invalid_unsigned_int);
+    __mmask8      mask    = _mm256_cmpneq_epu32_mask(invalid, index);
```

Comment:
Just for the record, this function is not in `AVX512F` but in `AVX512VL`, so we probably want to guard this with an `ifdef`.

Reply:
Is https://en.wikichip.org/wiki/x86/avx-512 correct in that this is only a problem for KNL?